content/6_Advanced/String_Search.mdx

```diff
@@ -126,8 +126,7 @@ def pi(s: str) -> List[int]:
 
 ## Z Algorithm
 
-The **Z-Algorithm** is another linear time string comparison algorithm like KMP,
-but instead finds the longest common prefix of a string and all of its suffixes.
+<FocusProblem problem="finding-periods" />
```

Comment:
is this problem just like the most direct application out there
bc i don't see you talking about how we use it to solve this

Reply:
The z-algorithm's main purpose is string matching. It does the same thing like KMP. Instead it has this application: finding periods, that KMP can't.